types/styled-system/index.d.ts

```diff
@@ -409,7 +410,76 @@ export function flexBasis(...args: any[]): any;
  * Grid Layout
  */
 
-// TODO: Add grid values
+export type GridGapValue = number | string;
+export type ResponsiveGridGapValue = ResponsiveValue<GridGapValue>;
+
+export interface GridGapProps {
+    gridGap?: ResponsiveGridGapValue;
+}
+
+export function gridGap(...args: any[]): any;
+
+export interface GridRowGapProps {
+    gridRowGap?: ResponsiveGridGapValue;
+}
+
+export function gridRowGap(...args: any[]): any;
+
+export interface GridColumnGapProps {
+    gridColumnGap?: ResponsiveGridGapValue;
+}
+
+export function gridColumnGap(...args: any[]): any;
+
+export type GridCellValue = string;
+export type ResponsiveGridCellValue = ResponsiveValue<GridCellValue>;
+
+export interface GridColumnProps {
+    gridColumn?: ResponsiveGridCellValue;
+}
+
+export function gridColumn(...args: any[]): any;
+
+export interface GridRowProps {
+    gridRow?: ResponsiveGridCellValue;
+}
+
+export function gridRow(...args: any[]): any;
+
+export type GridAutoValue = string;
```

Comment:
Grid auto flow is actually a union type, so you could write it like:

```
export type GridAutoFlowValue =
    "row"
    | "column"
    | "dense"
    | "row dense"
    | "column dense"
    | "inherit"
    | "initial"
    | "unset";
```

Reply:
@zephraph we should start using that CSS styles dependency that was mentioned on the first PR asap to set a precedent for people to follow.